### ISSUE TITLE
Update webpack-merge to v5

### ISSUE THREAD

**Issue by filoxo** [single-spa/create-single-spa#147]
Update webpack-merge to v5. See https://github.com/survivejs/webpack-merge/commit/534351d5e7dda8f4202cb90bdf6b6a43574de41f

I only took a cursory look at the updates. One change that we are interested in is that webpack-merge changes how to merge configs, specifically that `smart` merging was removed. We use `smart` in the generated Webpack configs to merge single-spa configs with user's configs.

**Comment by filoxo** [comment]
The release notes weren't where I would have expected them to be. This issue seems to be the most comprehensive list of changes: https://github.com/survivejs/webpack-merge/issues/136

**Comment by filoxo** [comment]
After trying to use webpack-merge@5 on a separate project, I'm of the opinion that webpack-merge is no longer useful. Its been completely gutted to the point where its just better not to use it. Here is an example, using the code from their documentation: 

```js
const output = mergeWithCustomize(
  {
    customizeArray(a, b, key) {
      if (key === 'extensions') {
        return _.uniq([...a, ...b]);
      }

      // Fall back to default merging
      return undefined;
    },
    customizeObject(a, b, key) {
      if (key === 'module') {
        // Custom merging
        return _.merge({}, a, b);
      }

      // Fall back to default merging
      return undefined;
    }
  }
)(object1, object2, object3, ...);
```

You can implement the same with just plain lodash (or any other alternative) with just a few functions which I find more readable.

```js
import { mergeWith, uniq, isObject, merge } from 'lodash';

function customize(a, b, key) {
  /* equal to customizeArray */
  if (Array.isArray(a)) { 
    if (key === 'extensions') {
      // custom behavior
      return uniq([...a, ...b]);
    }
    // can explicitly define the default behavior, eg. concat arrays by default
    return a.concat(b)
  }
  /* equal to customizeObject */
  if(isObject(a)){
    if (key === 'module') {
      return mergeWith(a, b, someOtherCustomizeFn);
    }
    // If this function returns `undefined`, merging is handled by the method instead
    // https://lodash.com/docs/4.17.15#mergeWith
    // so this is the same as webpack-merge
    return;
  }
}

const output = mergeWith(object1, object2, object3, ..., customize)
```

I'd be happy to be proven wrong but this is the conclusion I've reached after trying to use the updated webpack-merge for an hour.

**Comment by joeldenning** [comment]
From the code sample you provided, I tend to agree that the new webpack merge is not very useful. One tool that Vue CLI uses that is pretty nice is https://www.npmjs.com/package/webpack-chain. However, my understanding is that webpack chain cannot create a webpack config fom an existing object, but must be used everywhere as the thing that starts the webpack config. This is originally why I chose webpack merge, instead. Additionally, plain object syntax is often easier than the webpack chain syntax for simple tasks.

Open to ideas here on what to migrate to.

**Comment by filoxo** [comment]
webpack-merge@5.2.0 just added an API that will help with the advanced use case of merging loaders, see: https://github.com/survivejs/webpack-merge#mergewithrules 
which would restore a lot of the benefits of the previous version. I tried it out on a local project and it works well. I think we'd be able to proceed with using the latest version. Here is an example of how to use it with single-spa-webpack-config https://github.com/filoxo/webpack-merge-test/blob/main/webpack.config.js#L4-L14

**Comment by joeldenning** [comment]
I started work on this in #216

**Comment by joeldenning** [comment]
this is now released in 2.0
